wp-content/plugins/core/src/Blocks/Types/Post_List/Post_List.php

```diff
@@ -48,110 +61,210 @@ public function add_fields() {
+			],
+		] );
+		$group->add_field(
+			new Field( self::NAME . '_' . self::QUERY_POST_TYPES, [
+				'type'     => 'select',
+				'label'    => __( 'Post Types', 'tribe' ),
+				'multiple' => true,
+				'ui'       => true,
+				'name'     => self::QUERY_POST_TYPES,
+				'choices'  => $this->get_post_types_labels(),
+			] )
+		)->add_field(
+			new Field( self::NAME . '_' . self::QUERY_LIMIT, [
+				'label'         => __( 'Limit', 'tribe' ),
+				'name'          => self::QUERY_LIMIT,
+				'min'           => 2,
+				'max'           => 10,
+				'step'          => 1,
+				'type'          => 'range',
+				'default_value' => 2,
+			] )
+		)->add_field(
+			new Field( self::NAME . '_' . self::QUERY_TAXONOMIES, [
+				'type'          => 'select',
+				'multiple'      => true,
+				'label'         => __( 'Filter by Taxonomies', 'tribe' ),
+				'ui'            => true,
+				'name'          => self::QUERY_TAXONOMIES,
+				'return_format' => 'value',
+				'choices'       => $this->get_taxonomies_for_post_types(),
+			] )
 		);
-//		//We need to loop through all public post types to get taxonomy terms that only relate to the post type.
-//		//In a perfect world, we could conditionally update this field with javascript but that would require
-//		//a custom field being written. This will work for now. Could be a future consideration.
-		foreach ( self::QUERY_POST_TYPES as $post_type ) {
-			$this->add_field( new Field( self::NAME . '_' . self::TAXONOMIES . '_' . $post_type, [
-				'label'             => __( 'Filter by Taxonomy Term', 'tribe' ),
-				'name'              => self::TAXONOMIES . '_' . $post_type,
-				'post_type'         => $post_type,
-				'type'              => 'advanced_taxonomy_selector',
-				'field_type'        => 'multiselect',
-				'return_value'      => 'object',
+
+		//We need to loop through all public taxonomies to get taxonomy terms that only relate to the post type.
+		foreach ( $this->get_taxonomies_for_post_types() as $name => $label ) {
+			$group->add_field( new Field( self::NAME . '_' . self::QUERY_TAXONOMIES . '_' . $name, [
+				'label'             => sprintf(
+					__( 'Filter by %s Terms', 'tribe' ),
+					$label
+				),
+				'name'              => self::QUERY_TAXONOMIES . '_' . $name,
+				'type'              => 'taxonomy',
+				'field_type'        => 'multi_select',
+				'taxonomy'          => $name,
+				'allow_null'        => false,
+				'add_term'          => false,
+				'save_terms'        => false,
+				'load_terms'        => false,
+				'return_format'     => 'object',
 				'conditional_logic' => [
 					[
 						[
-							'field'    => 'field_ ' . self::NAME . '_' . self::QUERY_TYPE,
-							'operator' => '==',
-							'value'    => self::QUERY_TYPE_AUTO,
-						],
-						[
-							'field'    => 'field_' . self::NAME . '_' . self::POST_TYPES,
-							'operator' => '==',
-							'value'    => $post_type,
+							'field'    => 'field_' . self::NAME . '_' . self::QUERY_TAXONOMIES,
+							'operator' => '==contains',
```

Comment:
Is "==contains" a real operator? Or is this supposed to be "contains"?

Reply:
Oh it's real. Because ACF.